cmd/go-judge/rest_executor/cmd_handler_test.go

```diff
@@ -0,0 +1,184 @@
+package restexecutor
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/criyle/go-judge/cmd/go-judge/model"
+	"github.com/criyle/go-judge/envexec"
+	"github.com/criyle/go-judge/worker"
+	"github.com/gin-gonic/gin"
+	"go.uber.org/zap/zaptest"
+	"io"
+	"maps"
+	"net/http"
+	"net/http/httptest"
+	"slices"
+	"testing"
+	"time"
+)
+
+// mockWorker is a mock implementation of the worker.Worker interface
+type mockWorker struct {
+	// The result to send back when Submit is called
+	Result worker.Result
+	worker.Worker
+}
+
+func (m *mockWorker) Submit(_ context.Context, req *worker.Request) (<-chan worker.Response, <-chan struct{}) {
+	// Mock implementation
+	rtCh := make(chan worker.Response)
+	go func() {
+		rtCh <- worker.Response{
+			RequestID: req.RequestID,
+			Results:   []worker.Result{m.Result},
+		}
+	}()
```

Comment:
You don't need a go routine for this as you can create a buffered channel:

```go
rtCh := make(chan worker.Response, 1)
rtCh <- worker.Response{...
}
```

Also, do we want to check the request like what we did for results (`assertWorkerResultEqualsModelResult`) to ensure the api endpoint is parsing the request correctly?